packages/react-native-reanimated/Common/cpp/reanimated/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -265,10 +183,10 @@ jsi::Value NativeReanimatedModule::registerEventHandler(
       rt, worklet, "[Reanimated] Event handler must be a worklet.");
   int emitterReactTagInt = emitterReactTag.asNumber();
 
-  uiScheduler_->scheduleOnUI([=] {
+  NativeWorkletsModule_->getUIScheduler()->scheduleOnUI([=, this] {
```

Comment:
How about exposing `scheduleOnUI` directly from `NativeWorkletsModule`? Actually, we could do the same for `scheduleOnJS`.

Reply:
Good idea, but let's save it for the follow-up PR.